tangelo/linq/target/target_qiskit.py

```diff
@@ -92,11 +96,34 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
             job_sim = self.qiskit.execute(translated_circuit, backend, noise_model=qiskit_noise_model,
                                           shots=self.n_shots, basis_gates=None, optimization_level=opt_level)
             sim_results = job_sim.result()
-            frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
+            self.all_frequencies = {state[::-1]: count/self.n_shots for state, count in sim_results.get_counts(0).items()}
 
-            self.all_frequencies = frequencies.copy()
+            frequencies = self.all_frequencies
             self._current_state = None
 
+        # desired_meas_result is not None and return_statevector is requested so loop shot by shot (much slower)
+        elif desired_meas_result is not None:
+            from tangelo.linq.noisy_simulation.noise_models import get_qiskit_noise_model
+            n_meas = source_circuit.counts.get("MEASURE", 0)
+            backend = self.AerSimulator(method='statevector')
+            qiskit_noise_model = get_qiskit_noise_model(self._noise_model) if self._noise_model else None
+            translated_circuit = self.qiskit.transpile(translated_circuit, backend)
+            translated_circuit.save_statevector()
+            samples = dict()
+            self._current_state = None
+
+            for _ in range(self.n_shots):
+                sim_results = backend.run(translated_circuit, noise_model=qiskit_noise_model, shots=1).result()
+                current_state = sim_results.get_statevector(translated_circuit)
+                measurement = next(iter(self.qiskit.result.marginal_counts(sim_results, indices=list(range(n_meas))).get_counts()))[::-1]
```

Comment:
One question here: how the `marginal_counts` function play with the `list(range(n_meas))`? With this code:
https://github.com/goodchemistryco/Tangelo/blob/3f354fe35c0a7e5ac9dc57f9bfaa124d06ad5b73/tangelo/linq/target/backend.py#L247-L249
I understand that the first `n_meas` bits in the bitstring are for the mid-circuit measurements. In the code above, as qiskit qubit ordering is reversed, should the indices be a list of `n_meas` integers from the last qubit index instead of `list(range(n_meas)))`?

Reply:
I think since we're using the `qiskit.result.marginal_counts` functionality here to return results, they should be the right qubits. Afterwards, they are reversed with `[::-1]` to ensure the right ordering.